### PR TITLE
Fixes bullets floating in the air 

### PR DESCRIPTION
**Pull request by drexample**
## About The Pull Request

Port from Onyx's bay

## Why It's Good For The Game

Ay mlya ya masleenoo poymal

## Changelog
:cl:
fix: Bullets should no longer be able to float in the air without moving
/:cl:

### CODE REVIEW

File: code/modules/projectiles/projectile.dm [hunk 1]
```diff
@@ -838,7 +842,12 @@
 	return Process(targloc)
 
 /obj/item/projectile/test/Process(turf/targloc)
+	var/i = 0
 	while(src) //Loop on through!
+		if (++i > 512)
```
Comment (AlexMorgan3817) [inline code/modules/projectiles/projectile.dm:847]: Should be 513 or current DM version c:

File: code/modules/projectiles/projectile.dm [hunk 2]
```diff
@@ -684,9 +684,13 @@
 	return TRUE
 
 /obj/item/projectile/Process()
-	var/first_step = TRUE
-
+	var/first_step = 1
+	var/i = 0
 	spawn while(src && src.loc)
+		if (++i > 512)
```
Comment (mattroks101) [inline code/modules/projectiles/projectile.dm:690]: Why 512?

File: code/modules/projectiles/projectile.dm [hunk 3]
```diff
@@ -684,9 +684,13 @@
 	return TRUE
 
 /obj/item/projectile/Process()
-	var/first_step = TRUE
-
+	var/first_step = 1
```
Comment (mattroks101) [inline code/modules/projectiles/projectile.dm:687]: Why was this changed?

File: code/modules/projectiles/projectile.dm (hunk 2 above)
Comment (drexample) [reply code/modules/projectiles/projectile.dm:690]: Worked for Onyx, a direct copypaste from their fix

File: code/modules/projectiles/projectile.dm (hunk 3 above)
Comment (drexample) [reply code/modules/projectiles/projectile.dm]: Copypaste mistake, fixed

**Comment by MadmanMartian** [comment]
doesn't kill_count already do something similar to this?

**Review by MadmanMartian — CHANGES_REQUESTED** [review]
Confirm this actually does something, as right now it's just what kill_count does, but kill_count calls first as it is 50 process ticks versus 512 process ticks.

If it does, bring down the projectile_stuck required process ticks from 512 to something more reasonable.
